src/bin/juliainstaller.rs

```diff
@@ -99,7 +99,19 @@ struct Juliainstaller {
     juliaup_channel: String,
     /// Disable confirmation prompt
     #[clap(short = 'y', long = "yes")]
-    disable_confirmation_prompt: bool
+    disable_confirmation_prompt: bool,
+    /// Specify alternate path for Juliaup
+    #[clap(short = 'p', long = "path", default_value = "")]
+    alternate_path: String,
+    /// Control adding the Juliaup dir to the PATH
+    #[clap(long = "add-to-path", default_value = "yes")]
+    add_to_path: String,
+    /// Manually specify the background self-update interval
+    #[clap(long = "background-selfupdate", default_value = "0")]
+    background_selfupdate_interval: String,
+    /// Manually specify the statup self-update interval
+    #[clap(long = "startup-selfupdate", default_value = "1440")]
+    startup_selfupdate_interval: String
```

Comment:
Is there a way to use more specific types than `String` here? And have `clap` do the basic validation and type conversion?

Reply:
It's been a while, but I've spent this time ;earning more about rust and clap -- so this is me taking another stab at it.